File: system-test/java/src/test/java/com/mycompany/myshop/systemtest/legacy/mod11/base/BaseScenarioDslTest.java
package com.mycompany.myshop.systemtest.legacy.mod11.base;

import com.mycompany.myshop.systemtest.configuration.BaseConfigurableTest;
import com.mycompany.myshop.testkit.dsl.core.usecase.UseCaseDsl;
import com.mycompany.myshop.testkit.dsl.core.ScenarioDslImpl;
import com.mycompany.myshop.testkit.dsl.port.ScenarioDsl;
import com.mycompany.myshop.testkit.common.Closer;
import com.mycompany.myshop.systemtest.infrastructure.playwright.BrowserLifecycleExtension;
import com.optivem.testing.extensions.ChannelExtension;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith({ChannelExtension.class, BrowserLifecycleExtension.class})
public class BaseScenarioDslTest extends BaseConfigurableTest {
    private UseCaseDsl app;
    protected ScenarioDsl scenario;

    @BeforeEach
    void setUp() {
        var configuration = loadConfiguration();
        app = createUseCaseDsl(configuration);
        scenario = new ScenarioDslImpl(app);
    }

    @AfterEach
    void tearDown() {
        Closer.close(app);
    }
}

File: system-test/dotnet/SystemTests/Latest/Base/BaseScenarioDslTest.cs
using Dsl.Port;
using SystemTests.TestInfrastructure.Configuration;
using Dsl.Core;
using Optivem.Testing;
using Xunit;
using Dsl.Core.Scenario;

namespace SystemTests.Latest.Base;

public abstract class BaseScenarioDslTest : BaseConfigurableTest, IAsyncLifetime
{
    private UseCaseDsl _app = null!;

    public virtual async Task InitializeAsync()
    {
        var configuration = LoadConfiguration();
        _app = new UseCaseDsl(configuration);
        await Task.CompletedTask;
    }

    protected IScenarioDsl Scenario(Channel channel)
    {
        return new ScenarioDsl(channel, _app);
    }

    protected IScenarioDsl Scenario()
    {
        return new ScenarioDsl(null, _app);
    }

    public virtual async Task DisposeAsync()
    {
        if (_app != null)
            await _app.DisposeAsync();
    }
}












